SpokestackTests/NLUTensorflowSlotParserTest.swift

```diff
@@ -80,10 +80,16 @@ class NLUTensorflowSlotParserTest: XCTestCase {
     }
     
     func testParseUnspecifiedSlot() {
+        // Slots not declared by an intent but tagged by the model do not cause an error and are not returned to the caller
         let taggedInput = ["o", "b_eponymous"]
         let et = EncodedTokens(tokensByWhitespace: ["hey", "ma"], encodedTokensByWhitespaceIndex: [0, 1], encoded: [])
         let parsed = try! parser.parse(tags: taggedInput, intent: metadata!.intents.filter({ $0.name == "i.i" }).first!, encoder: encoder!, encodedTokens: et)
-        XCTAssertNil(parsed)
+        XCTAssertNil(parsed!["iMi"]!.value)
+        
+        // Slots declared by an intent but not tagged by the model are returned to the caller in the output with a null value
+        let taggedInputEmpty = ["o", "o"]
+        let parsedEmpty = try! parser.parse(tags: taggedInputEmpty, intent: metadata!.intents.filter({ $0.name == "i.i" }).first!, encoder: encoder!, encodedTokens: et)
+        XCTAssertNil(parsedEmpty!["iMi"]!.value)
```

Comment:
The number of exclamation points in the last two lines here is horrifically amusing.

Reply:
Swifty!